.github/workflows/main.yml

```diff
@@ -0,0 +1,55 @@
+on:
+  pull_request:
+    types: [opened]
+  issue_comment:
+    types: [created]
+
+jobs:
+  code_review_job:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+      contents: write
+    name: Run code review agent on every pull request, respond to user comments
+
+    steps:
+      - name: Set Options Environment Variable for Issue Comment
+        if: github.event_name == 'issue_comment'
+        run: |
+          GIT_DOMAIN_OPTION=""
+          if [[ -n "${{ vars.GIT_DOMAIN }}" ]]; then
+            GIT_DOMAIN_OPTION=" --git.domain=${{ vars.GIT_DOMAIN }}"
+          fi
+          STATIC_ANALYSIS_TOOL_OPTION=""
+          if [[ -n "${{ vars.STATIC_ANALYSIS_TOOL }}" ]]; then
+            STATIC_ANALYSIS_TOOL_OPTION=" --static_analysis_tool=${{ vars.STATIC_ANALYSIS_TOOL }}"
+          fi
+          echo "OPTIONS=--static_analysis.fb_infer.enabled=True --code_feedback=True --dependency_check.enabled=False --bee.path=/automation-platform  --bee.actn_dir=/automation-platform/default_bito_ad/bito_modules --git.access_token=${{ secrets.GIT_ACCESS_TOKEN }} --bito_cli.bito.access_key=${{ secrets.BITO_ACCESS_KEY }}${GIT_DOMAIN_OPTION}${STATIC_ANALYSIS_TOOL_OPTION}" >> $GITHUB_ENV
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/Amruta101998/DjangoProject/pull/2#issuecomment-2161011006>#059472</a><span> - 06/11/2024, 03:16 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The environment variable 'OPTIONS' is being set with multiple concatenated strings, including secrets and other variables. This approach can lead to issues if any of the variables are not set or contain unexpected values. Additionally, the use of '>>' to append to the environment file can cause issues if the file already contains data.
</div>
<div id="fix">
<b>Fix:</b> Use a more robust method to construct the 'OPTIONS' variable and ensure all required variables are set before appending to the environment file. Consider using a single 'echo' command with proper variable checks.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div>
<pre><code>
 @@ -28,7 +28,11 @@
           GIT_DOMAIN_OPTION=""
           if [[ -n "${{ vars.GIT_DOMAIN }}" ]]; then
             GIT_DOMAIN_OPTION=" --git.domain=${{ vars.GIT_DOMAIN }}"
           fi
           STATIC_ANALYSIS_TOOL_OPTION=""
           if [[ -n "${{ vars.STATIC_ANALYSIS_TOOL }}" ]]; then
             STATIC_ANALYSIS_TOOL_OPTION=" --static_analysis_tool=${{ vars.STATIC_ANALYSIS_TOOL }}"
           fi
 -         echo "OPTIONS=--static_analysis.fb_infer.enabled=True --code_feedback=True --dependency_check.enabled=False --bee.path=/automation-platform  --bee.actn_dir=/automation-platform/default_bito_ad/bito_modules --git.access_token=${{ secrets.GIT_ACCESS_TOKEN }} --bito_cli.bito.access_key=${{ secrets.BITO_ACCESS_KEY }}${GIT_DOMAIN_OPTION}${STATIC_ANALYSIS_TOOL_OPTION}" >> $GITHUB_ENV
 +         OPTIONS="--static_analysis.fb_infer.enabled=True --code_feedback=True --dependency_check.enabled=False --bee.path=/automation-platform --bee.actn_dir=/automation-platform/default_bito_ad/bito_modules --git.access_token=${{ secrets.GIT_ACCESS_TOKEN }} --bito_cli.bito.access_key=${{ secrets.BITO_ACCESS_KEY }}${GIT_DOMAIN_OPTION}${STATIC_ANALYSIS_TOOL_OPTION}"
 +         if [[ -z "${{ secrets.GIT_ACCESS_TOKEN }}" || -z "${{ secrets.BITO_ACCESS_KEY }}" ]]; then
 +           echo "Required secrets are not set" >&2
 +           exit 1
 +         fi
 +         echo "OPTIONS=${OPTIONS}" >> $GITHUB_ENV
</code></pre>
</div>
</details>
</div>
<br>


</div>